src/ember-app.js

```diff
@@ -285,13 +303,19 @@ class EmberApp {
   async _visit(path, fastbootInfo, bootOptions, result, buildSandboxPerVisit) {
     let shouldBuildApp = buildSandboxPerVisit || this._applicationInstance === undefined;
 
-    let app = shouldBuildApp ? await this._getNewApplicationInstance() : this._applicationInstance;
+    // if queue is empty we know sandbox will be eagerly built when using `buildSandboxPerVisit
+    let isSandboxEagerlyBuilt = this._sandboxApplicationInstanceQueue.isEmpty();
+
+    let app = shouldBuildApp ? await this.getNewApplicationInstance() : this._applicationInstance;
 
     if (buildSandboxPerVisit) {
       // entangle the specific application instance to the result, so it can be
       // destroyed when result._destroy() is called (after the visit is
       // completed)
       result.applicationInstance = app;
+
+      // we add analytics information about the current request to know where it is degraded or not.
+      result.analytics.sandboxPerVisit.isDegradedRequest = isSandboxEagerlyBuilt;
```

Comment:
I don't think "degraded" has the right meaning here. There are _many_ reasons a given request may have a degraded processing time, this is only representing one of them.

Some possible alternate names:

* `result.analytics.sandboxPerVisit.handledByQueue`
* `result.analytics.sandboxPerVisit.sandboxPrebuilt`
* ???

Reply:
Thanks for the feedback. Agree with you. I ended up with `usedPrebuiltSandbox`